### PR TITLE
Auth0_login && dashboard

### PR DESCRIPTION
**Pull request by lbai001**
autho0 login signup

user profile from auth0

basic profile display
add fontawsome

code clean up

Add forum link to dashboard navigation

remove general profile information from dashboard

recover rebase errors

switches login/dashboard in nav

### CODE REVIEW

File: components/DashboardNav/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  login() {
+    this.props.auth.login();
+  }
+
+  logout() {
+    this.props.auth.logout();
+  }
+
```
Comment (RobAWilkinson) [inline components/DashboardNav/index.js]: Please remove if not needed 👌

File: components/DashboardNav/styles.css [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  height: 100%;
+  position: relative;
+}
+.nav_item div {
+  cursor: pointer;
+  line-height: 50px;
+  width: 200px;
+}
+.nav_item div:hover {
+  color: tomato;
```
Comment (RobAWilkinson) [inline components/DashboardNav/styles.css]: match this with the color of the head on homepage

File: core/auth.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    scope: 'openid profile email'
+  });
+
+  constructor() {
+    this.login = this.login.bind(this);
+    this.logout = this.logout.bind(this);
+    this.handleAuthentication = this.handleAuthentication.bind(this);
+    this.isAuthenticated = this.isAuthenticated.bind(this);
+    this.getAccessToken = this.getAccessToken.bind(this);
+    this.getProfile = this.getProfile.bind(this);
```
Comment (RobAWilkinson) [inline core/auth.js]: lets move these into static class properties so we don't overdose on `bind` `login = () => { }` etc...

File: core/auth.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    browserHistory.push('/dashboard');
+  }
+
+  logout() {
+    // Clear access token and ID token from local storage
+    localStorage.removeItem('access_token');
+    localStorage.removeItem('id_token');
+    localStorage.removeItem('expires_at');
+    // navigate to the home route
+    browserHistory.push('/dashboard');
```
Comment (RobAWilkinson) [inline core/auth.js]: think about swapping with `replace` so the user doesn't go back to a page with no data upon logout

File: core/routes.js [hunk 5]
```diff
@@ -13,8 +13,17 @@ import listenerQuiz from '../experiments/listener-quiz/index';
...
 import ResultsContainer from '../pages/containers/ResultsContainer';
-
+import Dashboard from '../pages/dashboard/index';
+import Loading from '../pages/loading/index';
+import Auth from './auth';
 // this definitely needs work. it's a good start though.
+const auth = new Auth();
+const handleAuthentication = (nextState, replace) => {
+  if (/access_token|id_token|error/.test(nextState.location.hash)) {
+    console.log('hello?!', nextState.location.hash);
```
Comment (RobAWilkinson) [inline core/routes.js]: please remove console log

File: pages/containers/container.css [hunk 6]
```diff
@@ -9,6 +9,13 @@
   .wrap {
     min-height: calc(100vh - 225px);
     padding: 0px;
-    margin-top: 25px
+    margin-top: 25px;
   }
 }
+.login_link {
```
Comment (RobAWilkinson) [inline pages/containers/container.css]: dashes in css classes please `login-link`

File: core/auth.js [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+        // browserHistory.push('/dashboard');
+        this.getProfile(authResult.accessToken);
+      } else if (err) {
+        browserHistory.push('/error');
+        // location.replace('/error');
+        console.log(err);
+        alert(`Error: ${err.error}. Check the console for further details.`);
+      }
+    });
+  }
```
Comment (RobAWilkinson) [inline core/auth.js]: please convert into promise rather than callback

**Review by RobAWilkinson — CHANGES_REQUESTED** [review]

File: components/DashboardNav/index.js (hunk 1 above)
Comment (lbai001) [reply components/DashboardNav/index.js]: check

File: components/DashboardNav/styles.css (hunk 2 above)
Comment (lbai001) [reply components/DashboardNav/styles.css]: check

File: core/auth.js (hunk 3 above)
Comment (lbai001) [reply core/auth.js]: check

File: core/auth.js (hunk 7 above)
Comment (lbai001) [reply core/auth.js]: check

**Comment by RobAWilkinson** [comment]
LGTM 👍

**Pull request merged**